wordpress-mobile/WordPress-Android
# Move purgeMediaToPostAssociationIfNotInPostAnymore to UseCase

Merge instruction:
1. Review and merge https://github.com/wordpress-mobile/WordPress-Android/pull/10795 first
2. Change the target branch to develop or master-edit-post-activity-refactor

Moves purgeMedia.. into its own useCase and introduces unit tests.


To test:
1. Create a post
2. Turn on airplane mode
3. Add an image
4. Kill the app
5. Open the post - you can attach debugger or verify in Stetho that the MediaModel gets deleted
PR submission checklist:
